server/src/main/java/org/apache/druid/server/metrics/HistoricalMetricsMonitor.java

```diff
@@ -104,6 +104,32 @@ public boolean doMonitor(ServiceEmitter emitter)
       emitter.emit(builder.build("segment/count", count));
     }
 
+    for (Map.Entry<String, Long> entry : segmentLoadDropMgr.getAverageNumOfRowsPerSegmentForDatasource().entrySet()) {
+      String dataSource = entry.getKey();
+      long averageSize = entry.getValue();
+      final ServiceMetricEvent.Builder builder =
+          new ServiceMetricEvent.Builder().setDimension(DruidMetrics.DATASOURCE, dataSource)
+                                          .setDimension("tier", serverConfig.getTier())
+                                          .setDimension("priority", String.valueOf(serverConfig.getPriority()));
+
+      emitter.emit(builder.build("segment/averageRowCount", averageSize));
+    }
+
+    for (Map.Entry<String, SegmentRowCountBuckets> entry : segmentLoadDropMgr.getRowCountBucketsPerDatasource().entrySet()) {
+      String dataSource = entry.getKey();
+      SegmentRowCountBuckets rowCountBucket = entry.getValue();
+      final ServiceMetricEvent.Builder builder =
+          new ServiceMetricEvent.Builder().setDimension(DruidMetrics.DATASOURCE, dataSource)
+                                          .setDimension("tier", serverConfig.getTier())
+                                          .setDimension("priority", String.valueOf(serverConfig.getPriority()));
+
+      rowCountBucket.forEachDimension((final String bucketDimension, final int count) -> {
+        builder.setDimension("rowCountBucket", bucketDimension);
```

Comment:
Since `NA` is an error state, let's only report the metric if count > 0.

If you accept my suggestion below, I think the range `0` should also only be reported if the count is non-0.

Reply:
makes sense